pkg/certsigner/signer.go

```diff
@@ -77,14 +78,17 @@ type CertSigner struct {
 	cfsslSigner *local.Signer
 }
 
+// CertKey stores files for the cert and key pair.
+type CertKey struct {
+	CertFile, KeyFile string
+}
+
 // Config holds the configuration values required to start a new signer
 type Config struct {
 	// SignerCAFiles
 	SignerCAFiles
-	// ServerCertFile is the file location of the server certificate
-	ServerCertFile string
-	// ServerKeyFile is the file location of the server private key
-	ServerKeyFile string
+	// ServerCertKeys is a list of server certificates for serving on TLS based on SNI
+	ServerCertKeys []CertKey
```

Comment:
nit: Using `[]tls.Certificate` here (and loading the cert/key pairs in `cmd/kube-etcd-signer-server/serve.go`) would let us simplify the `pkg/certsigner` API slightly by leaning on stdlib types instead of defining our own.  But whatever, folks are unlikely to be using `pkg/certsigner` outside of this repository ;).